### ISSUE TITLE
Automatic response to incoming voice calls

### ISSUE THREAD

**Issue by razaar** [mautrix/whatsapp#536]
I'm getting quite a lot WhatsApp calls lately and I'm tired of explaining everyone why I can't answer (I think that's because WA changed the 'call contact' button in their app to use their servers and not the phone dialer). I'm not using WA on my phone so I miss all the calls and just see the 'Incoming call' message.

Would it be possible to implement sending an automatic answer via WhatApp to the contact who is calling, e.g. _'Sorry, only phone calls please!_'?

**Comment by jhaan1979** [comment]
Agreed. And preferably with the possibility to set your own reply, for instance if you want to offer directions to an alternative way of calling.

**Comment by xpseudonym** [comment]
Hi Folks, I posted a sort of work around idea on [ESS Community chat](https://matrix.to/#/!RkXKfjpGXIVObpCqOI:element.io/$JcZu7dKloy5aW8JHXr62Y3ZQTuEA1dN_ck7j_y9tTT0?via=matrix.org&via=element.io&via=read.nz)/ [ess-helm issue](https://github.com/element-hq/ess-helm/issues/731#issue-3413812887) and was told I was in the wrong place.  And indeed, given the OP, this does seem the correct place.  This is the basic idea:

> **Is your feature request related to a problem? Please describe.**
When a WhatsApp, Signal, Messenger, etc user tries to call me, instead of connecting the call ther call rings out and I get an irritating message like 'Incoming call. Use the WhatsApp app to answer.' (Substitute your particular chat app.)
> 
> This originated as a bit of star gazing on [the ESS Community chat room](https://matrix.to/#/!RkXKfjpGXIVObpCqOI:element.io/$JcZu7dKloy5aW8JHXr62Y3ZQTuEA1dN_ck7j_y9tTT0?via=matrix.org&via=element.io&via=read.nz).
> 
> **Describe the solution you'd like**
Mautrix bridges reply to calls with a text chat link to initiate an Element Call
> 
> **Additional context**
Pretty straight forward feature request - reply to calls with text link to an Element Call. I'm guessing the politics involved might make it more complicated than it appears at first look.

Basically, when I once got an irritating message 'Incoming call. Use the WhatsApp app to answer.' now my caller gets a 'click to talk' link to make me an Element Call.  I guess one reason this might not have occurred to the OP is that Element Call wasn't around when he first posted.

**Comment by kikislater** [comment]
That's a nice idea to answer to another call service !

**Comment by kikislater** [comment]
> I think it would be great if the sender receive an automatic message "Call me by phone"

Not convenient as most of the time in some countries, Whatsapp is used to not pay mobile call but to pay mobile data. And calling in a remote country is expensive as well
